infra/main.bicep

```diff
@@ -318,7 +318,8 @@ module formRecognizer 'core/ai/cognitiveservices.bicep' = {
   params: {
     name: !empty(formRecognizerServiceName) ? formRecognizerServiceName : '${abbrs.cognitiveServicesFormRecognizer}${resourceToken}'
     kind: 'FormRecognizer'
-    location: formRecognizerResourceGroupLocation
+    // Limited regions for new version: https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/concept-layout?view=doc-intel-4.0.0
+    location: 'eastus'
```

Comment:
Hmm, this might be a problem for merging. What do you think about conditionally avoiding the new file types if it's not east us?

Reply:
What about customers in Europe? Would be good to not set this sample hardcoded to East US, when there is also an option in Europe.

> The preview updates are available in only a few select regions that include US East, West US2 and West Europe.  The API version is 2023-10-31-preview. Check for the updated SDK in the [documentation](https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/changelog-release-history?view=doc-intel-4.0.0&tabs=csharp).

Reply:
That's fair, that means we do need to prompt for the location then. Or I could add a condition to select WestEurope automatically if the main location is closer to Europe but that doesn't feel particularly elegant.

Reply:
I updated the PR to prompt for location and to use a different naming prefix, so if you run it with an existing environment, it should make a whole new resource.

![Screenshot of location prompt](https://github.com/Azure-Samples/azure-search-openai-demo/assets/297042/c2046f87-2547-4d7d-ba61-b257a20a549d)